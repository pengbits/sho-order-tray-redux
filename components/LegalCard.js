import React, { Component } from 'react'
import {findDOMNode} from 'react-dom';
import {fromTo,set} from 'gsap';
import EasePack from  'gsap';
import Settings from '../settings'

class LegalCard extends Component {
  constructor(props){
    super(props)
    const {opacity} = props
    
    this.state = {
      opacity
    }
  }
  
  componentDidMount() {
    this.el = $(findDOMNode(this))
  }
  
  componentDidUpdate(prevProps, prevState) {
    //this.tween('enter')
  }
  
  componentWillAppear(callback){
    // this is for the case where we go from mobile -> desktop
    // with a provider selected... componentWillEnter doesn't fire,
    // but this does, so we can tween the card up from {opacity:0}
    this.tween('enter', callback)
    callback()
  }
  
  componentWillEnter(callback){
    this.tween('enter', callback)
  }
  
  componentWillLeave(callback){
    this.tween('leave', callback)
  }
  
  render(){
    const {type,content} = this.props;
    const style = {opacity: (this.props.opacity || 0)}
    
    return (<figure className="order-tray__legal-card" style={style}>
      <figcaption data-legal-type={type}>
        {content.map(this.renderParagraph)}
      </figcaption>
    </figure>
    )
  }
  
  renderParagraph(text,i) {
    return (<p key={i}>
      {text}
    </p>
    )
  }
  
  // note: we don't currently fade legal cards out,
  // but if we ever want to, passing mode of 'leave' should suffice
  tween(mode,callback){
    const isEnter  = mode == 'enter'
    const duration = isEnter ? Settings.legal_fade_up_duration : Settings.legal_fade_down_duration;
    const delay    = isEnter ? Settings.legal_delay : 0

    fromTo(
      this.el,
      duration,
      {
        opacity : (isEnter ? 0 : 1)
      },{
        opacity : (isEnter ? 1 : 0), 
        delay, 
        ease    : Settings.ease, 
        onComplete: () => {
          if(!!callback) callback()
        }
      }
    )
  }
}

export default LegalCard