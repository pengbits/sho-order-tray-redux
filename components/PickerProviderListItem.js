import cn from 'classnames';
import React, { Component } from 'react';

class PickerProviderListItem extends Component   {

  render () {
    const {onClick,groupId,isSelected,groupName} = this.props;
  
    return (
      <li className="order-picker__provider order-picker__control" data-context="order tray">
          <span
            onClick={onClick}
            id={`order-picker__provider-${groupId}`}
            data-provider-id={`${groupId}`} data-provider-group-id={`${groupId}`}
            className={cn('input','input--faux-radio', {'input--faux-radio-checked' : isSelected})}
          >
            <img className="input__icon" src={`https://www.sho.com/www/sho/lib/assets/svg/super-radio${isSelected ? '-checked':''}_36x36.svg`} /> 
            {groupName}
          </span>
      </li>
    )
  }
}
export default PickerProviderListItem;
