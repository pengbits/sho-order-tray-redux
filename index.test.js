import React from 'react';
import { shallow, mount } from 'enzyme';

import PickerProviderListItem from './components/PickerProviderListItem'
import LegalCard from './components/LegalCard'

describe('Order Tray Redux', () => {
	describe('Components', () => {
		describe('LegalCard', () => {
			test('renders LegalCard', () => {
				expect(shallow(<LegalCard
					type='STREAMING_PROVIDER'
					content={['do','the','right','thing']}
				/>)).toHaveLength(1)
			})
		})
		
		describe('Picker', () => {
			// these shallow render tests were failing 
			// when the variation mixin was still in place -
			// if we we need to restore that functionality,
		 // consider just removing the tests 
			describe('PickerProviderListItem', () => {
				const shallowRenderComponent = ({isSelected}={}) => {
					return shallow(<PickerProviderListItem 
						groupId={92}
						isSelected={!!isSelected}
						groupName='Apple' 
					/>);
				}
				
				test('renders', () => {
					let el = shallowRenderComponent()
					expect(el).toHaveLength(1)
				})
				
				test('matches snapshot', () => {
					let el = shallowRenderComponent()
					expect(el).toMatchSnapshot()
				})
				
				test('icon path and classnames change with selected state', () => {
					
					let el = shallowRenderComponent({isSelected: false});
					expect(el.find('.input--faux-radio').hasClass('input--faux-radio-checked')).toBe(false)
					expect(el.find('.input__icon').props().src).toMatch(/super-radio_36x36.svg$/);
					
					el = shallowRenderComponent({isSelected: true});
					expect(el.find('.input--faux-radio-checked')).toHaveLength(1)
					expect(el.find('.input__icon').props().src).toMatch(/super-radio-checked_36x36.svg$/);
					
				})
			})
		})
	})
})