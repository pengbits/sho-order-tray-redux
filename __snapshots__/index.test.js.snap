// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Tray Redux Components Picker PickerProviderListItem matches snapshot 1`] = `
<li
  className="order-picker__provider order-picker__control"
  data-context="order tray"
>
  <span
    className="input input--faux-radio"
    data-provider-group-id="92"
    data-provider-id="92"
    id="order-picker__provider-92"
  >
    <img
      className="input__icon"
      src="https://www.sho.com/www/sho/lib/assets/svg/super-radio_36x36.svg"
    />
    Apple
  </span>
</li>
`;
